6 tasks completed

ustinovsa opened the pull request.

ustinovsa committed:
6 tasks ready
w/o Country class

ustinovsa committed:
6 tasks ready
parse exception
trouble with task6

ustinovsa committed:
6 tasks ready

konivex commented on src/ru/avalon/java/j20/labs/models/Country.java:
```diff
@@ -57,10 +62,18 @@ public String getName() {
-    }
+        String[] splittedText= new String[2];
+        if (text.matches("^[A-Z]{2}[\\:]{1}.+$")==false) {
+            throw new ParseException("Ha-Ha, classic! "+
```
В регулярном выражении квадратные скобки лишние, множитель "{1}" после проверки на ":" лишний. Сравнение с false можно заменить логическим оператором (каким?).

ustinovsa replied:
если текст не содержит ****
т.е. в нашем случае
if (!(text.matches("^A-Z{2}\\:.+$"))

konivex commented on src/ru/avalon/java/j20/labs/tasks/Task1.java:
```diff
@@ -53,8 +58,16 @@ public void run() throws IOException {
-    private String read(File file) throws IOException {
-        throw new UnsupportedOperationException("Not implement yet!");
+    private String read1(File file) throws IOException {
+        try ( InputStream iStr = new FileInputStream(file)) {
```
Метод не будет работать как ожидается.

ustinovsa replied:
не понял от слова совсем) хоть направление укажите)
или открыть доки по InputStream FileInputStream будет достаточно?

konivex commented on src/ru/avalon/java/j20/labs/tasks/Task2.java:
```diff
@@ -53,8 +57,14 @@ public void run() throws IOException {
+    private String read2(File file) throws IOException{
+         StringBuilder sb = new StringBuilder();
+        try( Reader reader = new FileReader(file)){
+           while(reader.ready()==true){
```
Сравнение с true лишнее.

ustinovsa replied:
ok

konivex commented on src/ru/avalon/java/j20/labs/tasks/Task2.java:
```diff
@@ -53,8 +57,14 @@ public void run() throws IOException {
      */
-    private String read(File file) throws IOException {
-        throw new UnsupportedOperationException("Not implement yet!");
+    private String read2(File file) throws IOException{
```
Метод не будет работать как ожидается.

konivex reviewed (commented):
Исправьте замечания.

ustinovsa committed:
Исправлены методы чтения в Task1, Task2.
Результат чтения/записи выводит кириллицу корректно только
при считывании через байты в правильной кодировке (Windows-1251)
как я понял, из-за дефолтной кодировки String в JAVA (UTF-16),
а для корректной кодировки надо либj сначала переводить файл,
либо переписывать все через массив байтов.

ustinovsa reviewed (commented):
@konivex сейчас работает как надо, в моем понимании)
Регулярное выражение исправил, таски 1 и 2 тоже.

konivex reviewed (approved):
Зачтено.
